src/btree/bt_compact.c

```diff
@@ -65,6 +65,34 @@ __compact_rewrite(WT_SESSION_IMPL *session, WT_REF *ref, int *skipp)
 }
 
 /*
+ * __compact_skip --
+ *	Return if it may be worthwhile compacting a file.
+ */
+static int
+__compact_skip(WT_SESSION_IMPL *session, int *skipp)
+{
+	WT_BTREE *btree;
+	WT_PAGE_INDEX *pindex;
+
+	*skipp = 1;
+
+	btree = S2BT(session);
+
+	/*
+	 * It's safe to look at the pindex of the root page here. Trees with
+	 * less than 4 level one pages aren't worth compacting.
+	 */
+	pindex = WT_INTL_INDEX_GET_SAFE(btree->root.page);
+	if (pindex->entries < 4)
```

Comment:
I guess trees with 4 pages off the root could be quite large, but that feels like a corner case. I'd be happy to drop this change if you don't like it.